adapter/analytics/analytics.go

```diff
@@ -236,16 +239,22 @@ func (m *manager) Close() {
 // uploadLoop runs a timer that periodically pushes everything in the buffer
 // directory to the server.
 func (m *manager) uploadLoop() {
-	t := time.NewTicker(m.collectionInterval)
+	backoff := util.NewExponentialBackoff(200*time.Millisecond, 2 * time.Minute, 2, true)
+	retry := time.After(0 * time.Millisecond)
+
 	for {
 		select {
-		case <-t.C:
+		case <-retry:
 			if err := m.uploadAll(); err != nil {
 				m.log.Errorf("Error pushing analytics: %s", err)
 			}
+			m.log.Infof("Attempted %d times", backoff.Attempt())
```

Comment:
this log will only happen if there is an error, correct? let's just add the attempt count to the error log above.